tools/emscripten.py

```diff
@@ -627,11 +651,15 @@ def create_tsd(metadata, embind_tsd):
     out += ';\n'
   out += '}\n'
   out += f'\n{embind_tsd}'
+  # Combine all the various exports.
   export_interfaces = 'WasmModule'
+  if settings.EXPORTED_RUNTIME_METHODS:
+    export_interfaces += ' & typeof RuntimeExports'
+  # Add in embind defintions.
   if embind_tsd:
     export_interfaces += ' & EmbindModule'
   out += f'export type MainModule = {export_interfaces};\n'
```

Comment:
Perhaps a seperate change but can we skip the `export` here when `MODULARIZE` is enabled?  (i.e. just export the default/factory and nothing else?)

Also, I think we should rename `MainModule` to `Module` here, but again a separate change I guess.

Reply:
Yes, long term I think we should remove the export. Currently, quite a few projects are using `MainModule`, so I think it'd be best to keep this for now and get people to migrate to the new way, then delete `MainModule`.